stack/tools/src/main/java/org/apache/usergrid/tools/ManagementUserAudit.java

```diff
@@ -0,0 +1,209 @@
+
+        Option row_key = OptionBuilder.withArgName( ROW_KEY ).hasArg().isRequired( false )
+                                      .withDescription( "row key to check against" ).create( ROW_KEY );
+        options.addOption( row_key );
+
+        return options;
+    }
+
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.apache.usergrid.tools.ToolBase#runTool(org.apache.commons.cli.CommandLine)
+     */
+    @Override
+    public void runTool( CommandLine line ) throws Exception {
+        startSpring();
+
+        logger.info( "Starting entity unique checker" );
+
+        EntityManager em = emf.getEntityManager( MANAGEMENT_APPLICATION_ID );
+
+        if ( line.getOptionValue( ( "file" ) ) == null ) {
+            if ( line.getOptionValue( "dup" ) != null ) {
+                String extractedEmail = line.getOptionValue( "dup" );
+                column_verification( em, extractedEmail, line );
+            }
+            else {
+                logger.error( "Need to have -file or -dup not both and certainly not neither." );
+            }
+        }
+        else {
+            ObjectMapper objectMapper = new ObjectMapper();
+
+            File jsonObjectFile = new File( line.getOptionValue( "file" ) );
+
+            JsonNode node = objectMapper.readTree( jsonObjectFile );
+
+            JsonNode users = node.get( "user" );
+
+            for ( JsonNode email : users ) {
+
+                String extractedEmail = email.get( "name" ).getTextValue();
+                column_verification( em, extractedEmail, line );
+            }
+            logger.info( "Completed logging successfully" );
+        }
+    }
+
+
+    private void column_verification( final EntityManager em, final String extractedEmail, CommandLine line )
+            throws Exception {
+        UUID applicationId = MANAGEMENT_APPLICATION_ID;
+        String collectionName = "users";
+        String uniqueValueKey = "email";
+        String uniqueValue = extractedEmail;
+
+
+        Object key = key( applicationId, collectionName, uniqueValueKey, uniqueValue );
+
+
+        List<HColumn<ByteBuffer, ByteBuffer>> cols =
+                cass.getAllColumns( cass.getApplicationKeyspace( applicationId ), ENTITY_UNIQUE, key, be, be );
+
+        if ( cols.size() == 1 ) {
+            UUID uuid = null;
+            for ( HColumn<ByteBuffer, ByteBuffer> col : cols ) {
+                uuid = ue.fromByteBuffer( col.getName() );
+            }
+            if ( em.get( uuid ) == null ) {
+                logger.error( "Email: {} with uuid: {} doesn't exist in ENTITY_PROPERTIES.", extractedEmail, uuid );
+            }
+            else {
+                logger.info( "Email: {}  with uuid: {} exists in ENTITY_PROPERTIES for ENTITY_UNIQUE", extractedEmail,
+                        uuid );
+            }
+        }
+        else {
+            if ( cols.size() == 0 ) {
+                logger.error( "Email: {} doesn't exist in ENTITY_UNIQUE.", extractedEmail );
```

Comment:
Can we make this a WARN?  It's okay if it's not there, we mostly want to see errors when it's there, but has no data in ENTITY_PROPERTIES.